### PR TITLE
Classes and flow

### PR DESCRIPTION
**Pull request by orhanna14**
Alright, here's a go at 
- moving grid out of Coordinate initialization
- updating `PlayerInput` to `PlayerTurn` to better reflect it's responsibility
- Organizing `PlayerTurn` methods to public and private
- Updating the `run` method to be more readable

LMK what you think!

### CODE REVIEW

File: lib/player_turn.rb [hunk 1]
```diff
@@ -11,29 +11,30 @@ def initialize(stdin, grid, printer)
...
+    until coordinates.valid?(grid)
+      ask_for_coordinate_and_print_error
     end
     coordinates.value
   end
 
-  def get_coordinate_and_print_error
-    get_coordinate_input
-    unless coordinates.valid?
+  def ask_for_coordinate_and_print_error
```
Comment (baccigalupi) [inline lib/player_turn.rb]: Oh nice name change!

File: lib/player_turn.rb [hunk 2]
```diff
@@ -11,29 +11,30 @@ def initialize(stdin, grid, printer)
...
+    initialize_input_as_a_coordinate
```
Comment (baccigalupi) [inline lib/player_turn.rb]: This one is kind of long and not totally understandable though. `get_user_coordinate` ?

File: lib/player_turn.rb [hunk 3]
```diff
@@ -11,29 +11,30 @@ def initialize(stdin, grid, printer)
...
+    unless coordinates.valid?(grid)
       printer.print_coordinates_error
     end
   end
 
-  def coordinates
-    @coordinates ||= NullCoordinates.new
+  def get_initial_input
```
Comment (baccigalupi) [inline lib/player_turn.rb]: Also the initial here doesn't make sense to me since you are getting the input after a bad coordinate in entered too.

File: spec/player_turn_spec.rb [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+require_relative "../lib/player_turn.rb"
 require_relative "../lib/coordinates.rb"
 require_relative "../lib/grid.rb"
 require_relative "../lib/printer.rb"
 
-RSpec.describe PlayerInput do
+RSpec.describe PlayerTurn do
   describe "#get_valid_coordinate" do
      it "repeatedly asks for valid coordinate, returns the value when valid" do
       invalid_format = StringIO.new
```
Comment (baccigalupi) [inline spec/player_turn_spec.rb]: I think this should just be `stdin` or `input_stream`. It's hard to imagine its role when it is named that way.

File: spec/player_turn_spec.rb [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
       grid = Grid.new
       printer = Printer.new(invalid_format)
-      player_input = PlayerInput.new(invalid_format, grid, printer)
+      player_turn = PlayerTurn.new(invalid_format, grid, printer)
       allow(invalid_format).to receive(:gets).and_return("A6", "B7", "A1")
-      coordinates = Coordinates.new(player_input.get_user_input, double(grid))
+      coordinates = Coordinates.new(player_turn.get_initial_input)
```
Comment (baccigalupi) [inline spec/player_turn_spec.rb]: Why is this line needed? `get_initial_input` should be a private method. But also, the gets mocking should setup the input stream to do what you want.

**Review by baccigalupi — APPROVED** [review]
I looked a bit harder at the test for PlayerTurn and I think it needs to change. The rest looks so much better. I think some of the naming in PlayerTurn could use some improvements, but that's pretty minor!

File: lib/player_turn.rb (hunk 3 above)
Comment (orhanna14) [reply lib/player_turn.rb]: True. Good point

File: spec/player_turn_spec.rb (hunk 5 above)
Comment (orhanna14) [reply spec/player_turn_spec.rb]: Ah, right. Whoops! Fixed.

File: spec/player_turn_spec.rb (hunk 4 above)
Comment (orhanna14) [reply spec/player_turn_spec.rb]: I think for now I'll go back to `PlayerInput`.
